pkg/core/transaction/input.go

```diff
@@ -39,3 +39,20 @@ func (in *Input) EncodeBinary(w io.Writer) error {
 func (in *Input) Size() int {
 	return in.PrevHash.Size() + 2 // 2 = sizeOf uint16
 }
+
+// InputIntersection finds the intersection between to input slice a, b.
+func InputIntersection(a, b []Input) []Input {
+	m := make(map[Input]bool)
+	c := []Input{}
+
+	for _, item := range a {
+		m[item] = true
+	}
+
+	for _, item := range b {
+		if _, ok := m[item]; ok {
+			c = append(c, item)
+		}
+	}
+	return c
+}
```

Comment:
@dauTT example - https://play.golang.org/p/3pIe3MEy-4e
there's no reallocs, and more simple..  what do you think?

Reply:
:+1: